components/camel-huaweicloud-smn/src/main/java/org/apache/camel/component/huaweicloud/smn/SimpleNotificationProducer.java

```diff
@@ -0,0 +1,366 @@
+            httpConfig.withProxyHost(clientConfigurations.getProxyHost())
+                    .withProxyPort(clientConfigurations.getProxyPort())
+                    .setIgnoreSSLVerification(clientConfigurations.isIgnoreSslVerification());
+
+            if (clientConfigurations.getProxyUser() != null) {
+                httpConfig.withProxyUsername(clientConfigurations.getProxyUser());
+                httpConfig.withProxyPassword(clientConfigurations.getProxyPassword());
+            }
+        }
+
+        BasicCredentials credentials = new BasicCredentials()
+                .withAk(clientConfigurations.getAuthenticationkey())
+                .withSk(clientConfigurations.getSecretKey())
+                .withProjectId(clientConfigurations.getProjectId());
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Building Smn client");
+        }
+
+        // building smn client object
+        SmnClient smnClient = SmnClient.newBuilder()
+                .withCredential(credentials)
+                .withHttpConfig(httpConfig)
+                .withEndpoint(clientConfigurations.getServiceEndpoint())
+                .build();
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Successfully initialized Smn client");
+        }
+        return smnClient;
+    }
+
+    /**
+     * validation of all user inputs before attempting to invoke a service operation
+     *
+     * @param  simpleNotificationEndpoint
+     * @param  exchange
+     * @return
+     */
+    private ClientConfigurations validateServiceConfigurations(
+            SimpleNotificationEndpoint simpleNotificationEndpoint, Exchange exchange) {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Inspecting exchange body");
+        }
+        // verifying if exchange has valid body content. this is mandatory for 'publish as text' operation
+        if (ObjectHelper.isEmpty(exchange.getMessage().getBody())) {
+            if (simpleNotificationEndpoint.getOperation().equals("publishAsTextMessage")) {
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Found null/empty body. Cannot perform publish as text operation");
+                }
+                throw new IllegalArgumentException("exchange body cannot be null / empty");
+            }
+        }
+
+        // checking for mandatory field 'operation name'
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Inspecting operation name");
+        }
+        if (ObjectHelper.isEmpty(exchange.getProperty("CamelHwCloudOperation"))
+                && ObjectHelper.isEmpty(simpleNotificationEndpoint.getOperation())) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Found null/empty operation name. Cannot proceed with Smn operations");
+            }
+            throw new IllegalArgumentException("operation name not found");
+        } else {
+            clientConfigurations.setOperation(exchange.getProperty("CamelHwCloudOperation") != null
```

Comment:
same here for the exchange properties constants

Reply:
Moved to SmnProperties class